examples/mobile/HomeApp/js/app.js

```diff
@@ -220,21 +312,25 @@ import Storage from "./clipping-storage.js";
 		burgerButton.addEventListener("click", onButtonClick);
 		popupButton.addEventListener("click", onPopupSubmit);
 
-		appsList = storage.readAllFromStorage(Storage.elements.APPSLIST) || [];
+		appsList = storage.readAllFromStorage(Storage.elements.APPSLIST);
+		if (appsList.length == 0) {
+			appsList = defaultList;
+		}
 
 		//TODO: all webclips url should checked if they are valid
 
-		if (!appsList.length) {
-			getAppsList.then((apps) => {
-				appsList = apps;
-				updateWebClipsUI();
-				updateWebClipListPopup();
-				storage.refreshStorage(Storage.elements.APPSLIST, apps);
-			});
-		} else {
+		if (appsList.length) {
 			updateWebClipsUI();
 			updateWebClipListPopup();
 		}
+		// check webclips on remote server
+		getAppsList.then((apps) => {
+			updateAppsList(apps);
+			storage.refreshStorage(Storage.elements.APPSLIST, appsList);
```

Comment:
you write to storage appsList but updateAppsList with apps

Reply:
This is properly. The method `updateAppsList` changes `appsList` variable so after update I save new state of `appsList` to storage

Reply:
ok, this was a little bit misleading for